src/client/replication_ddl_client.cpp

```diff
@@ -1738,5 +1738,27 @@ replication_ddl_client::set_max_replica_count(const std::string &app_name,
         configuration_set_max_replica_count_rpc(std::move(req), RPC_CM_SET_MAX_REPLICA_COUNT));
 }
 
+error_with<configuration_rename_app_response>
+replication_ddl_client::rename_app(const std::string &old_app_name, const std::string &new_app_name)
+{
+    if (old_app_name.empty() ||
+        !std::all_of(old_app_name.cbegin(),
+                     old_app_name.cend(),
+                     (bool (*)(int))replication_ddl_client::valid_app_char))
+        return FMT_ERR(ERR_INVALID_PARAMETERS, "invalid old_app_name.");
```

Comment:
It seems this code is duplicated some times, could you add a macros like:
```
VALIDATE_TABLE_NAME(table_name, msg)  \
  if (table_name.empty() || ...) {    \
    return FMT_ERR(ERR_INVALID_PARAMETERS, #msg);
```

you can do it in another patch and refactor some old code.